CCM-12017-Azure recommendations (#1767)
harness/developer-hub: docs/cloud-cost-management/4-use-ccm-cost-optimization/1-ccm-recommendations/azure-vm.md

---
title: Azure VM recommendations

> description: Optimize the utilization of your EC2 instances using Harness CCM recommendations.

Reviewer: Optimize the utilization of your "Azure VM" using Harness CCM recommendations.

Virtual machines are considered low-utilization:

* If their CPU utilization is 5 percent or less and their network utilization is less than 2 percent and have threshold memory pressure numbers

* If the current workload can be accommodated by a smaller virtual machine size.

You can view the recommendations for your Azure VMs on the **Recommendations** page. 

:::note
Before using recommendations in your environment, ensure that you evaluate their impact thoroughly. The person reviewing the recommendations should be able to understand the impacts identified in the recommendations, as well as the impact on the infrastructure and business.

Using recommendations without proper assessment could result in unexpected changes, such as issues with system performance or poor reliability.
:::

## Before you begin

* Connect your Azure cloud account in Harness and set up CCM for cost management. For more information, go to [Set up cost visibility for Azure](../../2-getting-started-ccm/4-set-up-cloud-cost-management/set-up-cost-visibility-for-azure.md).

> * To obtain Azure VM recommendations, configure a Harness Azure CCM connector with the Inventory Management feature enabled.

Reviewer: Inventory Management & Visibility(By Default) feature enabled.

Reviewer: The Visibility feature would enable us to get the recommendation from the Azure Advisor Portal. And the Inventory Management feature would enable us to get the CPU Utilization data and  show it with respect to recommendation. If Inventory Management feature not enabled we might see null state in the graph & the CPU Utilization of the recommendation.

## How are Azure recommendations computed?

The recommendations are computed by analyzing the past CPU utilization. Harness CCM leverages the Azure Advisor recommendations. CCM uses the Azure APIs to fetch data from the Azure account.

:::note
Azure Advisor automatically runs in the background to find newly created resources. It can take up to 24 hours to provide recommendations on those resources.
:::

### Configure Azure Advisor

You must configure Azure Advisor to receive recommendations for cost optimization. Go to [Enable Azure recommendations](../../2-getting-started-ccm/4-set-up-cloud-cost-management/set-up-cost-visibility-for-azure.md).

## Types of Azure VM recommendations

Harness CCM provides two types of recommendations to optimize your VMs:

> * **Resizing**: In this type of recommendation, CCM recommends resizing your VM based on usage. For information about the different types of Azure VMs, go to [Available instance types](https://learn.microsoft.com/en-us/azure/virtual-machines/sizes). 

Reviewer: Vm Rightsizing

---
title: Azure VM recommendations
description: Optimize the utilization of your Azure VMs using Harness CCM recommendations.
# sidebar_position: 2
---

# Optimize Azure VM costs with recommendations

An effective way to reduce Azure VM costs is to optimize VM utilization. This involves resizing idle or underutilized VMs based on active tasks and shutting down unused VMs.

> Virtual machines are considered low-utilization:

Reviewer 2: VMs?

Reviewer 2: Global comment

:::note
Azure Advisor automatically runs in the background to find newly created resources. It can take up to 24 hours to provide recommendations on those resources.
:::

### Configure Azure Advisor

You must configure Azure Advisor to receive recommendations for cost optimization. For more information, go to [Enable Azure recommendations](../../2-getting-started-ccm/4-set-up-cloud-cost-management/set-up-cost-visibility-for-azure.md).

## Types of Azure VM recommendations

Harness CCM provides two types of recommendations to optimize your VMs:

* **VM Rightsizing**: In this type of recommendation, CCM recommends resizing your VM based on usage. For information about the different types of Azure VMs, go to [Available instance types](https://learn.microsoft.com/en-us/azure/virtual-machines/sizes). 

* **Shutdown**: In this type of recommendation, the instance is terminated or shutdown if not in use for a long time.  
  
## View your Azure recommendations

1. In **Harness**, go to the **Cloud Costs** module.

> 2. Click **Recommendations**.

Reviewer 2: ```suggestion
2. Select **Recommendations**.
```

> 3. Click the filter icon.

Reviewer 2: ```suggestion
3. Select the filter icon.
```

4. In the **Recommendation Type** dropdown list, select **AZURE_INSTANCE**.

> 5. Click **Apply**. 

Reviewer 2: ```suggestion
5. Select **Apply**. 
```

>   The recommendations for Azure VMs are displayed. 

Reviewer 2: ```suggestion
  The recommendations for Azure VMs appear.
```

Reviewer 2: just to keep it active

Author: Displayed is more appropriate. it just filters.

> 6. Click the row to view the recommendations for that account.  The CPU utilization graph shows the current utilization data. Go to [View and apply recommendations](https://developer.harness.io/docs/cloud-cost-management/use-cloud-cost-management/ccm-recommendations/home-recommendations#enable-azure-recommendations) to learn how to apply these recommendations.

Reviewer 2: ```suggestion
6. Select a row to view the recommendations for that account.  
    
    The CPU utilization graph shows the current utilization data. Go to [View and apply recommendations](https://developer.harness.io/docs/cloud-cost-management/use-cloud-cost-management/ccm-recommendations/home-recommendations#enable-azure-recommendations) to learn how to apply these recommendations.
```
